sim_levy_iqr.R

```diff
@@ -1,29 +1,155 @@
-
-iqr_tab <- iqr_tab %>%
-  rowwise() %>%
-  mutate(IQR = qstable(Top, a, b, g, d) - qstable(Bottom, a, b, g, d)) %>% # find the relevant IQRs for the given parameters
-  ungroup() %>%
-  mutate(pc_IQR = ifelse((row_number() %% 2) == 0, (IQR - lag(IQR,1))/lag(IQR,1), NA),
-         pc_IQR = round(100*pc_IQR, digit = 2)) # express as %
-
-print(xtable(iqr_tab, caption = 'Interquartile Range as a Measure of Dispersion'), include.rownames = FALSE) # paste table into latex
\ No newline at end of file
+# This file runs the simulation for Interquartile Ranges
+
+rm(list = ls()) # clear environment
+gc() # garbage collection
+
+pacman::p_load(tidyverse,xtable,stabledist,plotly) # load packages
+
+iqr_tab <- as.data.frame(rbind(c(0.95,0.05,1.5,0,1,0), # create the parameters for the relevant Levy distributions
+                               c(0.95,0.05,1.3,0,1,0),
+                               c(0.9,0.1,1.5,0,1,0),
+                               c(0.9,0.1,1.3,0,1,0),
+                               c(0.75,0.25,1.5,0,1,0),
+                               c(0.75,0.25,1.3,0,1,0),
+                               c(0.95,0.05,1.3,0,1,0),
+                               c(0.95,0.05,1.3,1,1,0),
+                               c(0.9,0.1,1.3,0,1,0),
+                               c(0.9,0.1,1.3,1,1,0),
+                               c(0.75,0.25,1.3,0,1,0),
+                               c(0.75,0.25,1.3,1,1,0)))
+colnames(iqr_tab) <- c('Top', 'Bottom', 'a', 'b', 'g', 'd')
+
+iqr_tab <- iqr_tab %>%
+  rowwise() %>%
+  mutate(IQR = qstable(Top, a, b, g, d) - qstable(Bottom, a, b, g, d)) %>% # find the relevant IQRs for the given parameters
+  ungroup() %>%
+  mutate(pc_IQR = ifelse((row_number() %% 2) == 0, (IQR - lag(IQR,1))/lag(IQR,1), NA), # we want to compute the pairwise differences (row 2 vs 1, 4 vs 3 - not 3 vs 2!) - the ifelse checks whether the row is even before computing a number
+         pc_IQR = round(100*pc_IQR, digit = 2)) # express as %
+
+print(xtable(iqr_tab, caption = 'Interquartile Range as a Measure of Dispersion'), include.rownames = FALSE) # paste table into latex
+# Scale of distribution is affected by gamma - direct impact on IQR, but alpha and beta can also change it
+
+# Plot IQR as a function of beta
+# First set parameter values
+
+a <- 1.5
+g <- 1
+d <- 0
+Top <- 0.75
+Bottom <- 0.25
+
+IQR <- list()
+
+for(i in 0:200){
+  b <- (i - 100)/100 # these are the possible beta values
+  IQR[i+1] <- qstable(Top, a, b, g, d) - qstable(Bottom, a, b, g, d)
+}
+
+# Create plot
+
+pdf('C:/Users/JulianW/Documents/Work/Productivity Dispersion/sim_graphs/beta_IQR.pdf', height = 6, width = 8)
+
+plot(range(beta),
+     range(IQR),
+     type = 'n',
+     cex = 0.9,
+     xlab = 'Beta', 
+     ylab = 'IQR',
+     cex.axis = 1.3,
+     cex.lab = 1.3)
+lines(beta,
+      IQR,
+      lty = 1,
+      col = 'steelblue',
+      lwd = 2)
+
+dev.off()
+
+# Surface plots - how does the IQR change for choices of beta, alpha, and the relevant precentiles

```

Comment:
The 3 following computations seem quite repetitive. Could you not just compute `IQR_1`, `IQR_2`, and `IQR_3` in the same loop? Or, alternatively, more abstractly make this a function and create a list of IQR matrices (but this might be too much abstraction).